examples/energy-management-app/energy-management-common/src/EnergyEvseTargetsStore.cpp

```diff
@@ -0,0 +1,478 @@
+
+        ChipLogProgress(AppServer, "LoadTargets: DayOfWeekForSequence = 0x%02x",
+                        mChargingTargetSchedulesArray[chargingTargetSchedulesIdx].dayOfWeekForSequence.GetField(
+                            static_cast<TargetDayOfWeekBitmap>(0x7F)));
+
+        // ChargingTargets List
+        ReturnErrorOnFailure(reader.Next(TLV::kTLVType_List, TLV::ContextTag(TargetEntryTag::kChargingTargetsList)));
+        TLV::TLVType chargingTargetsListType;
+        ReturnErrorOnFailure(reader.EnterContainer(chargingTargetsListType));
+
+        // The mChargingTargets object handles the allocation of the chargingTargets. Let it know the currentSchedule index
+        mChargingTargets.Reset(chargingTargetSchedulesIdx);
+
+        // Load the chargingTargets associated with this schedule
+        while ((err = reader.Next(TLV::kTLVType_Structure, TLV::ContextTag(TargetEntryTag::kChargingTargetsStruct))) ==
+               CHIP_NO_ERROR)
+        {
+            TLV::TLVType chargingTargetsStructType = TLV::kTLVType_Structure;
+            ReturnErrorOnFailure(reader.EnterContainer(chargingTargetsStructType));
+
+            // Keep track of the current chargingTarget being loaded
+            EnergyEvse::Structs::ChargingTargetStruct::Type chargingTarget;
+
+            while ((err = reader.Next()) == CHIP_NO_ERROR)
+            {
+                auto type = reader.GetType();
+                auto tag  = reader.GetTag();
+                if (type == TLV::kTLVType_NotSpecified)
+                {
+                    // Something wrong - we've lost alignment
+                    return CHIP_ERROR_UNEXPECTED_TLV_ELEMENT;
+                }
+
+                if (tag == TLV::ContextTag(TargetEntryTag::kTargetTime))
+                {
+                    ReturnErrorOnFailure(reader.Get(chargingTarget.targetTimeMinutesPastMidnight));
+                }
+                else if (tag == TLV::ContextTag(TargetEntryTag::kTargetSoC))
+                {
+                    chip::Percent tempSoC;
+                    ReturnErrorOnFailure(reader.Get(tempSoC));
+                    chargingTarget.targetSoC.SetValue(tempSoC);
+                }
+                else if (tag == TLV::ContextTag(TargetEntryTag::kAddedEnergy))
+                {
+                    int64_t tempAddedEnergy;
+                    ReturnErrorOnFailure(reader.Get(tempAddedEnergy));
+                    chargingTarget.addedEnergy.SetValue(tempAddedEnergy);
+                }
+                else
+                {
+                    // Something else unexpected here
+                    return CHIP_ERROR_UNEXPECTED_TLV_ELEMENT;
+                }
+            }
+
+            ReturnErrorOnFailure(reader.ExitContainer(chargingTargetsStructType));
+
+            [[maybe_unused]] int64_t addedEnergy = chargingTarget.addedEnergy.HasValue() ? chargingTarget.addedEnergy.Value() : 0;
```

Comment:
use `ValueOr(0)` for better readability here and below.

Reply:
Change made.